### ISSUE TITLE
Dart SDK Version issue

### ISSUE THREAD

**Issue by DhavalRKansara** [shamblett/iot-home#4]
Hi Shamblett,

While trying to get the packages, I am getting the following error:

**Error:** `Because iot_home depends on grove any which requires SDK version >=2.6.0 <3.0.0, version solving failed.`

and the same for the **mraa** dependency.

**Comment by shamblett** [comment]
Well spotted, I've updated the SDK constraints, try it now.

**Comment by DhavalRKansara** [comment]
I think you misunderstood me.
My previous comment meant that few of the dependencies( mraa, grove) used in your project had a higher configuration of dart SDK.
Currently, the latest stable available version of Dart is 2.5.0, and because of that, I am unable to use your project.

**Comment by shamblett** [comment]
Latest stable version of Dart is 2.6.1, see [here](https://dart.dev/tools/sdk/archive)

**Comment by DhavalRKansara** [comment]
Thanks, Shamblett

it is working fine and I am closing this issue.

**Comment by DhavalRKansara** [comment]
With Dart SDK version 2.6.1 facing the below issue.

> Wrong full snapshot version, expected 'cb187597a166f2c0c64a200e22509ee3' found 'c8562f0ee0ebc38ba217c7955956d1cb' 
> 
> Isolate creation failed

**Comment by shamblett** [comment]
Did you create the snapshot with dart 2.5.0? I'm not sure what you are trying to do here.

**Comment by DhavalRKansara** [comment]
The latest stable version of flutter is compatible with dart 2.5.0.

We can upgrade the dart with `flutter upgrade` command only or is there any other way to upgrade dart SDK?

I tried to download the latest stable release of Flutter SDK(2.6.1) from the [link](https://dart.dev/tools/sdk/archive) and replace it with the already available dart-sdk inside **flutter > bin > cache > dart-sdk** but after this steps, I am getting below error:
```

Wrong full snapshot version, expected 'cb187597a166f2c0c64a200e22509ee3' found 'c8562f0ee0ebc38ba217c7955956d1cb' 

Isolate creation failed
```

**Comment by shamblett** [comment]
I'm not a flutter user unfortunately, you may be better asking on a flutter list.

**Comment by DhavalRKansara** [comment]
@shamblett  Which framework did you use?

Would it be possible to change the dart SDK version to 2.1.0  so that this project can be used in a flutter.

**Comment by shamblett** [comment]
I just use the Dart VM and its SDK, mraa and grove depend on FFI which itself needs Dart 2.6 or above which is where we started, I'm surprised Flutter is so far behind Dart stable, there were a few releases between Dart 2.5.0 and Dart 2.6. Could you not use Flutter dev channel?

**Comment by bhagujava** [comment]
@shamblett The issue is current flutter stable release support dart<=2.5.0 and one of your **Mraa** repo contain `environment:
 sdk: ">=2.6.0  <3.0.0" `

**Comment by DhavalRKansara** [comment]
> I just use the Dart VM and its SDK, mraa and grove depend on FFI which itself needs Dart 2.6 or above which is where we started, I'm surprised Flutter is so far behind Dart stable, there were a few releases between Dart 2.5.0 and Dart 2.6. Could you not use Flutter dev channel?

@shamblett  No, I am using the stable channel of the flutter not a dev one.

**Comment by DhavalRKansara** [comment]
> > I just use the Dart VM and its SDK, mraa and grove depend on FFI which itself needs Dart 2.6 or above which is where we started, I'm surprised Flutter is so far behind Dart stable, there were a few releases between Dart 2.5.0 and Dart 2.6. Could you not use Flutter dev channel?
> 
> @shamblett No, I am using the stable channel of the flutter not a dev one.

I already raised a question on [StackOverflow](https://stackoverflow.com/questions/59170895/how-to-update-dart-sdk-to-latest-stable-version-in-flutter) for the same issue.

**Comment by shamblett** [comment]
I'm not sure what you mean, as I said mraa uses the new Dart [FFI ](https://dart.dev/guides/libraries/c-interop)language feature which is only available in Dart 2.6 or above, it can't use any lower SDK becaus e it won't work, same with the grove package.

**Comment by DhavalRKansara** [comment]
@shamblett  Got it. Thanks for your support.

I will give a try on the flutter dev channel and let's see what happens.
